src/meta/src/manager/catalog/fragment.rs

```diff
@@ -449,25 +545,96 @@ where
+                    reschedule.removed_actors.iter().cloned().collect();
+
+                for (upstream_fragment_id, dispatcher_id) in
+                    reschedule.upstream_fragment_dispatcher_ids
+                {
+                    let upstream_fragment = table_fragment
+                        .fragments
+                        .get_mut(&upstream_fragment_id)
+                        .unwrap();
+                    for upstream_actor in &mut upstream_fragment.actors {
+                        for dispatcher in &mut upstream_actor.dispatcher {
+                            if dispatcher.dispatcher_id == dispatcher_id {
+                                dispatcher.hash_mapping =
+                                    Some(reschedule.upstream_dispatcher_mapping.clone());
+                                let mut new_downstream_actor_ids = reschedule.added_actors.clone();
+                                for actor_id in &dispatcher.downstream_actor_id {
+                                    if !removed_actor_ids.contains(actor_id) {
+                                        new_downstream_actor_ids.push(*actor_id);
+                                    } else {
+                                        tracing::debug!(
+                                            "remove downstream {} from actor {} dispatcher {}",
+                                            actor_id,
+                                            upstream_actor.actor_id,
+                                            dispatcher.dispatcher_id
+                                        );
+                                    }
+                                }
+
+                                dispatcher.downstream_actor_id = new_downstream_actor_ids;
+                            }
+                        }
+                    }
+                }
+
+                if let Some(downstream_fragment_id) = reschedule.downstream_fragment_id {
+                    let downstream_fragment = table_fragment
+                        .fragments
+                        .get_mut(&downstream_fragment_id)
+                        .unwrap();
+                    for downstream_actor in &mut downstream_fragment.actors {
+                        let mut new_upstream_actor_ids = reschedule.added_actors.clone();
+                        for actor_id in &downstream_actor.upstream_actor_id {
+                            if !removed_actor_ids.contains(actor_id) {
+                                new_upstream_actor_ids.push(*actor_id);
+                            } else {
+                                tracing::debug!(
+                                    "remove upstream {} from actor {}",
+                                    actor_id,
+                                    downstream_actor.actor_id
+                                );
+                            }
+                        }
+                        downstream_actor.upstream_actor_id = new_upstream_actor_ids;
```

Comment:
Looks like this logic is the same as that in `update_merge_node_upstream`. Can we merge them?